### PR TITLE
Update ChaCha20-Poly1305 cipher suites

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
Updates cipher suites to most recent RFC so interops with BoringSSL, OpenSSL, and GNUTLS.
```
ECDHE-RSA-CHACHA20-POLY1305
DHE-RSA-CHACHA20-POLY1305
ECDHE-ECDSA-CHACHA20-POLY1305
```

Introduces -OLD cipher suites for interop with LibreSSL and www.google.com server with chacha-poly.
```
ECDHE-RSA-CHACHA20-POLY1305-OLD
DHE-RSA-CHACHA20-POLY1305-OLD
ECDHE-ECDSA-CHACHA20-POLY1305-OLD
```

Fix use of supported curves extension with ChaCha suite using ECC.

Change naming scheme for AEAD macros to be more explicit.

Fix free type in FreeArrays src/internal.c and clear both Arrays and Keys struct after session.

Add population of supported curves extension to client.c (needed for ECDHE suites to connect with BoringSSL server).

Add API wc_Poly1305_MAC to poly1305.c for easy generation of recent padding scheme MACs and add a test of API to test.c.

### CODE REVIEW

File: wolfcrypt/src/poly1305.c [hunk 1]
```diff
@@ -550,5 +560,81 @@ int wc_Poly1305Update(Poly1305* ctx, const byte* m, word32 bytes) {
...
+    }
+    paddingLen = -sz & (WC_POLY1305_PAD_SZ - 1);
+    if (paddingLen) {
+        if ((ret = wc_Poly1305Update(ctx, padding, paddingLen)) != 0) {
+            return ret;
+        }
+    }
+
+    /* size of additional data and input as little endian 64 bit types */
+    U32TO64(addSz, little64);
```
Comment (dgarske) [inline wolfcrypt/src/poly1305.c]: Hey Jacob, what happens if we are on a BIG_ENDIAN_ORDER system? Is it possible you'll need to use a macro like LITTLE32 in chacha.c which makes sure addSz and sz (below) are little endian?

**Comment by JacobBarthelmeh** [comment]
Hi David, special treatment of addSz and sz is not needed.

Have tested this on our Big Endian power pc system. 
Little Endian mac client (both wolfSSL/OpenSSL) to Big Endian power pc server (wolfSSL) connection.
Big Endian power pc client to Big Endian power pc server(both wolfSSL/OpenSSL) connection
Big Endian power pc client to Little Endian mac server connection

**Comment by dgarske** [comment]
Awesome! Just wanted to double check on that. Great work on the code! Reviewed it in detail and couldn't see anything else. Do any of the commit tests perform external tests to validate this works with both styles of Chacha/Poly AEAD? It looks like it does, but wanted to confirm.

**Comment by JacobBarthelmeh** [comment]
Thanks for reviewing it. There is no commit script in place for external testing but a manual test can be done by running the command
```
./examples/client/client -g -d -p 443 -h www.google.com -l ECDHE-RSA-CHACHA20-POLY1305-OLD
```
I'm not sure of a public facing server using the more recent RFC. For me tests of switching between the two where done by interoperability tests with BoringSSL, which also supports both old and new RFC's.

**Comment by dgarske** [comment]
Is there a way we can add the above command to the external test?

**Pull request merged**